.github/workflows/testing-and-deployment.yml

```diff
@@ -149,21 +149,22 @@ jobs:
       # see discussion at https://github.com/pyvista/pyvista/issues/2867
       matrix:
         include:
+          # numeric numpy versions are < conditions, e.g. "1.24" means "numpy<1.24"
           - python-version: "3.8"
             vtk-version: "9.0.3" # Requires numpy<1.24
-            numpy-version: "default"
+            numpy-version: "1.24"
           - python-version: "3.9"
             vtk-version: "9.1"
-            numpy-version: "default"
+            numpy-version: "2.0" # matplotlib and numpy incomptability for python 3.9
```

Comment:
```suggestion
            numpy-version: "2.0" # matplotlib and numpy incompatibility for python 3.9
```

Reply:
I think I might get rid of this comment altogether, as it is also nice to have testing for numpy 1.26.  With the comment, someone might figure out the versioning and remove this pin.